client/lib/mention.js

```diff
@@ -0,0 +1,67 @@
+var hueHash = require('./hue-hash')
+
+/**
+ * Determines if name contains the characters in the partial name, in order.
+ */
+module.exports.containsSubseq = function(name, part) {
+  var offset = 0
+  for(var partOffset = 0; partOffset < part.length; partOffset++) {
+    var nextChar = part.substr(partOffset, 1)
```

Comment:
Any reason you chose to use `substr` instead of `part[partOffset]`? I was curious about perf since this code'll be pretty hot, here's a comparison: http://jsperf.com/substr-vs-slicing